src/next-app/pages/quote-request.tsx

```diff
@@ -0,0 +1,328 @@
+import DropdownField from '../components/DropdownField';
+import TextMultiField from '../components/TextMultiField';
+import SingleDropdownField from '../components/SingleDropdownField';
+import carData from '../data/data.json';
+import validateEmail from '../utils/validateEmail';
+import TextArea from '../components/TextArea';
+import ShopCard from '../components/ShopCard';
+// import { ImagePicker } from 'react-file-picker';
+
+interface carModels {
+    [make: string]: string[],
+}
+
+interface checkedShops {
+    [shop: string]: boolean
+}
+
+const QuoteRequestPage: NextPage = () => {
+    const [name, setName] = useState('');
+    const [make, setMake] = useState('');
+    const [customMake, setCustomMake] = useState('');
+    const [model, setModel] = useState('');
+    const [modelYear, setModelYear] = useState('');
+    const [customModel, setCustomModel] = useState('');
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const [email, setEmail] = useState('');
+    const [emailErrors, setEmailErrors] = useState<Set<string> | undefined>(undefined);
+    const [preferredContact, setPreferredContact] = useState('');
+    const [notes, setNotes] = useState('');
+    const [partPreferenceSeller, setPartPreferenceSeller] = useState('');
+    const [partPreferenceType, setPartPreferenceType] = useState('');
+    const [startDate, setStartDate] = useState('');
+    const [endDate, setEndDate] = useState('');
+    const [open, setOpen] = useState(false);
+    const [checkedShops, setCheckedShops] = useState({} as checkedShops);
+    const [disableSubmitShops, setDisableSubmitShops] = useState(true);
+    const [checkedShopsDisplay, setCheckedShopsDisplay] = useState([] as JSX.Element[]);
+
+    let makes: string[] = [];
+    let models: carModels = {};
+
+    let valid = false;
+    if (name.length < 1 ||
+        (make.length < 1 || customMake.length < 1) ||
+        (model.length < 1 || customModel.length < 1) ||
+        modelYear.length < 1 ||
+        firstName.length < 1 || lastName.length < 1 ||
+        phoneNumber.length < 1 ||
+        email.length < 1 ||
+        (emailErrors && emailErrors.size > 0) ||
+        checkedShopsDisplay.length === 0
+    )
+
+        carData.carData.forEach((entry) => {
+            makes.push(entry.make);
+            models[entry.make] = [];
+            models[entry.make] = entry.models.sort((a, b) => {
+                if (a == 'Other') {
+                    return 1;
+                }
+                if (b == 'Other') {
+                    return -1;
+                }
+                if (a > b) {
+                    return 1;
+                }
+                if (a < b) {
+                    return -1
+                }
+                return 0
+            });
+        });
+
+    makes.sort();
+    makes.push('Other');

```

Comment:
This logic for parsing and sorting the car data is run on every re-render, whereas it only needs to be run once on page mount. Consider putting it inside a `useEffect`.